src/pykiso/test_coordinator/test_execution.py

```diff
@@ -179,6 +230,28 @@ def enable_step_report(all_tests_to_run: unittest.suite.TestSuite) -> None:
             setattr(tc, method_name, assert_decorator(method))
 
 
+def parse_test_selection_pattern(pattern: str) -> List[Optional[str]]:
+    """Parse test selection pattern from cli.
+    For example: test_file.py::test_class::test_case
+
+    :param pattern: test selection pattern
+    :return: pattern for file, class name and test case name
+    """
+    if not pattern:
+        return [None] * 3
+    parsed_patterns = []
+    patterns = pattern.split("::")
+    for pattern in patterns:
+        if pattern == "":
+            parsed_patterns.append(None)
+        else:
+            parsed_patterns.append(pattern)
+    for _ in range(3 - (len(parsed_patterns))):
+        parsed_patterns.append(None)
+
+    return parsed_patterns
```

Comment:
Maybe make a nice namedtuple to return instead, something like 
```
TestFilterPattern = namedtuple("TestFilterPattern", "file, class, case")
```

and then l.310 you'd have something like:
```
filter_patterns = parse_test_selection_pattern(pattern_inject)
test_suites = collect_test_suites(config["test_suite_list"], filter_patterns.file)
...
if test_class_pattern:
    all_tests_to_run = apply_test_case_filter(
        all_tests_to_run, filter_patterns.class, filter_patterns.case
    )
```

Reply:
added tuple approach